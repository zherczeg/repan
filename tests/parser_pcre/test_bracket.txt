# Testing bracket types.

# Modifiers (i, m, s needs to be kept, the rest affects parse time only)

/(?imsxn)ab(?-imsxn)c/
    print

/(?s)(?m)(?i)abc/
    print

/(?xixmim)ab(?mmm-mmmsss)c/
    print

/(?xxmxi)ab(?sss-sss)c/
    print

/(?:(?s)a.b)/
    print

/(?i:(?m)(?s)(?-i)a.b)/
    print

/(?ims)a(?^)b(?i)c(?^i)d(?n)e/
    print

/(?ims:a(?^:b)c(?^)d(?^:ee)f(?m:gg(?m:hh)))i/
    print

/(?m:a(?i-m:b)c(?-im)d)/
    print

/(?i:a(?m:b)c(?-m)d)/
    print

/(?i-m-m:a)/
    print

/(?i-m-m)a/
    print

/(?^i-m)a/
    print

/a(?is/
    print

/a(?iX/
    print

# Non-capturing group

/(?n)(ab)(?<name>cd)(ef)\1/
    print

/(ab)(?n:(cd)(ef))(gh)/
    print

# Extended regexp

/(?x)   a
  b	\   c
   # comment
   d
   ### comment ###/
    print

/(?xx)[ a b	c \  d]/
    print

/a  (?x:  b	c   d)  e/
    print


# Other types of brackets except conditional brackets

/(?=pos_ahead)(?!neg_ahead)(?<=pos_behind)(?<!neg_behind)/
    print

/(?>atomic)/
    print

/a(?#comment\)b/
    print

/a(?#comment/
    print
