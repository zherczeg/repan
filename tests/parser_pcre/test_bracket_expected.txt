# Testing bracket types.

# Modifiers (i, m, s needs to be kept, the rest affects parse time only)

/(?imsxn)ab(?-imsxn)c/
    print
    /(?ims)ab(?-ims)c/

/(?s)(?m)(?i)abc/
    print
    /(?ims)abc/

/(?xixmim)ab(?mmm-mmmsss)c/
    print
    /(?im)ab(?-m)c/

/(?xxmxi)ab(?sss-sss)c/
    print
    /(?im)abc/

/(?:(?s)a.b)/
    print
    /(?s:a.b)/

/(?i:(?m)(?s)(?-i)a.b)/
    print
    /(?ms:a.b)/

/(?ims)a(?^)b(?i)c(?^i)d(?n)e/
    print
    /(?ims)a(?-ims)b(?i)cde/

/(?ims:a(?^:b)c(?^)d(?^:ee)f(?m:gg(?m:hh)))i/
    print
    /(?ims:a(?-ims:b)c(?-ims)d(?:ee)f(?m:gg(?:hh)))i/

/(?m:a(?i-m:b)c(?-im)d)/
    print
    /(?m:a(?i-m:b)c(?-m)d)/

/(?i:a(?m:b)c(?-m)d)/
    print
    /(?i:a(?m:b)cd)/

/(?i-m-m:a)/
    print
    Pattern parse failed: Invalid hypen in option list
        /(?i-m[ HERE ]>>-m:a)/

/(?i-m-m)a/
    print
    Pattern parse failed: Invalid hypen in option list
        /(?i-m[ HERE ]>>-m)a/

/(?^i-m)a/
    print
    Pattern parse failed: Invalid hypen in option list
        /(?^i[ HERE ]>>-m)a/

/a(?is/
    print
    Pattern parse failed: Unterminated option list
        /a[ HERE ]>>(?is/

/a(?iX/
    print
    Pattern parse failed: Unknown option in (?...) option list
        /a(?i[ HERE ]>>X/

# Non-capturing group

/(?n)(ab)(?<name>cd)(ef)\1/
    print
    /(?:ab)(?<name>cd)(?:ef)\g{1}/

/(ab)(?n:(cd)(ef))(gh)/
    print
    /(ab)(?:(?:cd)(?:ef))(gh)/

# Extended regexp

/(?x)   a
  b	\   c
   # comment
   d
   ### comment ###/
    print
    /ab cd/

/(?xx)[ a b	c \  d]/
    print
    /[abc d]/

/a  (?x:  b	c   d)  e/
    print
    /a  (?:bcd)  e/


# Other types of brackets except conditional brackets

/(?=pos_ahead)(?!neg_ahead)(?<=pos_behind)(?<!neg_behind)/
    print
    /(?=pos_ahead)(?!neg_ahead)(?<=pos_behind)(?<!neg_behind)/

/(?>atomic)/
    print
    /(?>atomic)/

/a(?#comment\)b/
    print
    /ab/

/a(?#comment/
    print
    Pattern parse failed: Unterminated comment
        /a[ HERE ]>>(?#comment/
